app/js/lib/server/settings.js

```diff
@@ -0,0 +1,236 @@
+'use strict';
+
+import EventDispatcher from '../common/event-dispatcher';
+
+// Prefix all entries to avoid collisions.
+const PREFIX = 'calendar-';
+
+const ORIGIN = 'https://calendar.knilxof.org';
+
+/**
+ * API version to use (currently not configurable).
+ * @type {number}
+ * @const
+ */
+const API_VERSION = 1;
+
+/**
+ * Regex to match upper case literals.
+ * @type {RegExp}
+ * @const
+ */
+const UPPER_CASE_REGEX = /([A-Z])/g;
+
+const p = Object.freeze({
+  values: Symbol('values'),
+  storage: Symbol('storage'),
+
+  // Private methods.
+  updateSetting: Symbol('updateSetting'),
+  stringToSettingTypedValue: Symbol('stringToSettingTypedValue'),
+  getDefaultSettingValue: Symbol('getDefaultSettingValue'),
+  onStorage: Symbol('onStorage'),
+});
+
+// Definition of all available settings and their default values (if needed).
+const settings = Object.freeze({
+  // String settings.
+  SESSION: Object.freeze({ key: 'session' }),
```

Comment:
Right, makes sense. I would advocate the 'type' to be a function - i.e. `Number` or `Boolean` are both functions that coerce the given value into the expected "`typeof`", 
e.g: `typeof Boolean(1) === 'boolean'` etc.  This way we can always run new values through this function to ensure the expected type - otherwise the 'type' is purely documentation from what I can see?